src/avutil.ml

```diff
@@ -42,6 +42,38 @@ module Time_format = struct
   ]
 end
 
+module Log = struct
+  type level = [
+    | `Quiet
+    | `Panic
+    | `Fatal
+    | `Error
+    | `Warning
+    | `Info
+    | `Verbose
+    | `Debug
+  ]
+
+  let set_level level =
+    ()
+
+  external set_log_callback : (string -> unit) -> unit = "ocaml_avutil_set_log_callback"
+
+  let set_callback fn =
+    let m = Mutex.create () in
+    let fn s =
+      Mutex.lock m;
```

Comment:
I get the impression that this mutex is redundant with the ocaml runtime system global lock, am I wrong ?

Reply:
The avutils doc says that the logging function should be thread safe as it's called from multiple threads concurrently. I'm actually not sure that the global lock can achieve this. The OCaml runtime isn't fully concurrent, indeed, but it may still switch between OCaml threads while they are being executed, I believe. Otherwise there wouldn't be a need for the `Mutex` module.. :-)